src/template.js

```diff
@@ -0,0 +1,76 @@
+import {Parser, tokTypes as tt} from "acorn";
+
+const SCOPE_FUNCTION = 2;
+const SCOPE_ASYNC = 4;
+const SCOPE_GENERATOR = 8;
+
+export default function parseTemplate(input) {
```

Comment:
Some comments here about why this needs to be hand written like that would help. I would have thought we could just use the Acorn parseTemplate function?